### ISSUE TITLE
Change resourceId to deviceId and path

### ISSUE THREAD

**Issue by poussa** [otcshare/iotivity-node#29]
Currently the API spec is :

Client
```
retrieveResource(USVString resourceId);
updateResource(USVString resourceId, OicResourceInit resource);
deleteResource(USVString resourceId); 
startObserving(USVString resourceId); 
```
Server
```
unregisterResource(USVString resourceId);
notify(USVString resourceId, OicMethod method,
```
The resourceId is an internally generated ID which is passed around and made available during the discovery. 

I would like the **resourceId** to be changed to tuple: device-id (di) and path (href), those are uniquely identifying the resources and can be passed to external entities as well like via REST API to the browser/cloud clients.

**Comment by zolkis** [comment]
Fixed by #30

**Comment by zqzhang** [comment]
This shall be fixed.
